efsity/src/main/kotlin/org/smartregister/structuremaptool/GenerateStructureMapService.kt

```diff
@@ -0,0 +1,254 @@
+package org.smartregister.structuremaptool
+
+import ca.uhn.fhir.context.FhirContext
+import ca.uhn.fhir.context.FhirVersionEnum
+import ca.uhn.fhir.parser.IParser
+import java.io.File
+import java.io.FileInputStream
+import java.nio.charset.Charset
+import java.util.*
+import org.apache.commons.io.FileUtils
+import org.apache.poi.ss.usermodel.CellType
+import org.apache.poi.ss.usermodel.Row
+import org.apache.poi.ss.usermodel.WorkbookFactory
+import org.hl7.fhir.r4.context.SimpleWorkerContext
+import org.hl7.fhir.r4.model.Bundle
+import org.hl7.fhir.r4.model.Parameters
+import org.hl7.fhir.r4.model.Questionnaire
+import org.hl7.fhir.r4.model.QuestionnaireResponse
+import org.hl7.fhir.utilities.npm.FilesystemPackageCacheManager
+import org.hl7.fhir.utilities.npm.ToolsVersion
+import org.smartregister.external.TransformSupportServices
+import org.smartregister.structuremaptool.extensions.addIndentation
+import org.smartregister.structuremaptool.extensions.clean
+
+fun main(
+  xlsConfigFilePath: String,
+  questionnaireFilePath: String,
+  questionnaireResponsePath: String
+) {
+  val contextR4 = FhirContext.forR4()
+  val fhirJsonParser = contextR4.newJsonParser()
+  val questionnaire: Questionnaire =
+    fhirJsonParser.parseResource(
+      Questionnaire::class.java,
+      FileUtils.readFileToString(File(questionnaireFilePath), Charset.defaultCharset())
+    )
+  val questionnaireResponse: QuestionnaireResponse =
+    fhirJsonParser.parseResource(
+      QuestionnaireResponse::class.java,
+      FileUtils.readFileToString(File(questionnaireResponsePath), Charset.defaultCharset())
+    )
+
+  // reads the xls
+  val xlsConfigFile = FileInputStream(xlsConfigFilePath)
+  val xlWorkbook = WorkbookFactory.create(xlsConfigFile)
+
+  // TODO: Check that all the Resource(s) ub the Resource column are the correct name and type eg.
+  // RiskFlag in the previous XLSX was not valid
+  // TODO: Check that all the path's and other entries in the excel sheet are valid
+  // TODO: Add instructions for adding embedded classes like
+  // `RiskAssessment$RiskAssessmentPredictionComponent` to the TransformSupportServices
+
+  // read the settings sheet
+  val settingsWorkbook = xlWorkbook.getSheet("Settings")
+  var questionnaireId: String? = null
+
+  for (i in 0..settingsWorkbook.lastRowNum) {
+    val cell = settingsWorkbook.getRow(i).getCell(0)
+    if (cell.stringCellValue == "questionnaire-id") {
+      questionnaireId = settingsWorkbook.getRow(i).getCell(1).stringCellValue
+    }
+  }
```

Comment:
I looked around and it seems like there's no search or column access, which is unfortunate. However we can make this more efficient with at least these 2 changes:

1. use [`.iterator`](https://poi.apache.org/apidocs/dev/org/apache/poi/ss/usermodel/Sheet.html#iterator--) in a foreach loop
2. most importantly, break the loop after you find the value